src/System.Net.Security/src/System/Net/Unix/SecuritySafeHandles.cs

```diff
@@ -19,24 +19,31 @@ internal sealed class SafeFreeCertContext : DebugSafeHandle
     internal sealed class SafeFreeCertContext : SafeHandle
     {
 #endif
-        public SafeFreeCertContext(IntPtr handle) : base(handle, true)
+        private readonly SafeX509Handle _certificate;
+
+        public SafeFreeCertContext(SafeX509Handle certificate) : base(IntPtr.Zero, true)
         {
+            if ((null != certificate) && !certificate.IsInvalid)
+            {
+                _certificate = Interop.libcrypto.X509_dup(certificate);
```

Comment:
You very much need to know what OpenSSL did/does.  I'm pretty sure the code as-is will leave the native certificate at the end with a tracking reference count of 1, so it won't be freed.  This means every SslStream leaks a native resource to the tune of an X509* and all of its references.